### DISCUSSION TITLE
How do I use the inline chat window instead of the input box?

### DISCUSSION THREAD

**Discussion by Pranith12250** [microsoft/vscode-discussions#1212] [Extension Development QnA] [answered]
I am been developing an extension that uses vscode.window.showInputBox() to get some basic input from the user regarding the code on the line of the active editor they are currently working on and then perform an appropriate command.

But this functionality would be much better with an inline input box exactly like the inline chat window used by GitHub copilot rather than a regular input box which opens the input box inline. I recently saw this feature being used by GitHub Copilot on a YouTube short posted by the official YouTube page of VS Code right  [here in this link](https://youtu.be/HiWR1RJHOg4?si=wNyyaGt2Ilpe5R7b)

So how do I move the input box from the top of the window to be inline as shown in the window? I cannot find any documentation for this either to help me out. If there is, please let me know. Is it even possible to get an inline input box?

**Comment by tamuratak** [comment, marked as answer]
You can refer to:

- https://code.visualstudio.com/api/extension-guides/chat
- https://github.com/microsoft/vscode-discussions/discussions/1210

The API is available on VS Code Insiders now.

The command of starting an inline chat is `inlineChat.start`.

**Reply by Pranith12250** [reply]
I read the reference you provided but I want to know if it is possible to use the inline chat for taking simple input from the user the way we can using vscode.window.showInputBox() rather than have a chat environment inline. The reference seems to talk more about setting up a chat system and I could not find any particular reference to having an inline chat window there.

I tried executing the command using vscode.commands.executeCommand('inlineChat.start'); in vs code insiders but that does not do anything. Is it possible to have an input box appear inline only by setting up a chat system and extending GitHub Copilot? Or could you give me an idea on how I could set up my own custom input box for taking simple input that appears inline? What tools would I have to use to create such an input box that appears inline?

**Reply by Rainmen-xia** [reply]
as I know, inline-chat cannot be use without github copilot. if you can ,please tell me

**Comment by theReynald** [comment]
@Pranith12250, Does that last response help?
